common/persistence/visibility/store/standard/sql/visibility_store.go

```diff
@@ -323,6 +323,26 @@ func (s *visibilityStore) CountWorkflowExecutions(
 	return nil, store.OperationNotSupportedErr
 }
 
+func (s *visibilityStore) GetWorkflowExecution(
+	ctx context.Context,
+	request *manager.GetWorkflowExecutionRequest,
+) (*store.InternalGetWorkflowExecutionResponse, error) {
+	rows, err := s.sqlStore.Db.SelectFromVisibility(ctx, sqlplugin.VisibilitySelectFilter{
+		NamespaceID: request.NamespaceID.String(),
+		RunID:       &request.RunID,
+	})
+	if err != nil {
+		return nil, serviceerror.NewUnavailable(
+			fmt.Sprintf("GetWorkflowExecution operation failed. Select failed: %v", err))
+	}
+	if len(rows) == 0 {
+		return &store.InternalGetWorkflowExecutionResponse{}, nil
```

Comment:
should this be NotFound error?

Reply:
Sure, I can change. I just thought it makes sense to return a response with nil execution.

Reply:
And the same for ES. It should `return nil, serviceerror.NotFound`.